llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -340,8 +341,33 @@ class ScalarizerVisitor : public InstVisitor<ScalarizerVisitor, bool> {
   const unsigned ScalarizeMinBits;
 };
 
+class ScalarizerLegacyPass : public FunctionPass {
```

Comment:
@boomanaiden154 and @nikic Since you removed and approved the removal of the legacy pass in this commit: https://github.com/llvm/llvm-project/commit/2470857fe701f41278dac33580ca8b0bdb38031f

I wanted to make sure you were ok with our reasons for bringing it back for the DIrectX backend?

Reply:
It's ok to bring the pass back for backend use. Just to confirm, the pass is a legalization requirement and injecting it via a pass builder callback into the middle-end pipeline is not an option?

Reply:
Yes its a legalization requirement. It handles extraneous inserts element cleanup as-well that we need for the DXIL to be correct. So it makes the most sense for us to do this in the backend.

Reply:
@nikic or @boomanaiden154 would you be willing to approve this PR.